packages/browser/src/core/user/index.ts

```diff
@@ -183,18 +200,92 @@ export interface CookieOptions {
   sameSite?: string
 }
 
+export class UniversalStorage {
+  private stores: Store[]
+
+  constructor(stores?: Store[]) {
+    this.stores = stores || []
+  }
+
+  private getStores(storeTypes: StoreType[] | undefined): Store[] {
+    return storeTypes
+      ? this.stores.filter((s) => storeTypes.indexOf(s.getType()) !== -1)
+      : this.stores
+  }
+
+  public push(store: Store) {
+    this.stores.push(store)
+  }
+
+  public getAndSync<T>(key: string, storeTypes?: StoreType[]): T | null {
```

Comment:
Something feels a bit off about this method -- it breaks command-query separation, but that's maybe not it (returning data in an update isn't that uncommon) -- maybe calling it `syncAndGet` would be a bit better, to show that its primary function should be to sync.

Reply:
Same, not a fan of this method either. The reason it exists is probably two folds: 
- If a value exists in one storage but not others ( for whatever reason, i.e., cookies are cleared but localStorage is not ) it syncs the identities across the storages again 
- There is some formatting mismatch between AJS classic and next cookies, and this makes sure that old AJS cookies are re-written in the new format as soon as we call `getAndSync` 

The reason not refactoring this is to reduce the blast reduce of this PR. This touches fairly critical part of the code, and don't want to break things in one huge PR.

Reply:
Gotcha, I kind of missed that this logic already existed -- all of this context would be great as a comment!